addons/website_sale_wishlist/migrations/12.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    group_public = env.ref('base.group_public')
+    partners = group_public.users.mapped('partner_id')
```

Comment:
I think you should add here also this, just in case there are multiple websites or they have different public users:

```suggestion
    partners = group_public.users.mapped('partner_id') | env['website'].search([]).mapped('partner_id')
```